### PR TITLE
Fix for R before 4

### PR DESCRIPTION
**Pull request by tuxmaster5000**
Fir for https://github.com/astamm/nloptr/issues/105

### CODE REVIEW

File: tools/cmake_call.sh [hunk 1]
```diff
@@ -13,12 +13,19 @@ cd src
 sh ./scripts/nlopt_download.sh ${RSCRIPT_BIN}
 dot() { file=$1; shift; . "$file"; }
 dot ./scripts/r_config.sh ""
+
+# Only R > 4 will have the variables
+$RSCRIPT_BIN --vanilla -e "getRversion() > 4" | grep TRUE > /dev/null
```
Comment (eddelbuettel) [inline tools/cmake_call.sh]: Can you make it `'getRversion() >= "4.0.0"'` which is more idiomatic and clearer?  (That is outer single quote ' protecting inner double quotes.)  A more elegant way would use the return code as `grep` is a little dirty.  But it does a proper version comparison and is better than grep'ing `Rscript --version`.

File: tools/cmake_call.sh [hunk 2]
```diff
@@ -13,12 +13,19 @@ cd src
...
+if [ $? -eq 0 ]; then
+	NEW_R="-D CMAKE_AR=${AR} -D CMAKE_RANLIB=${RANLIB} "
+else
+	NEW_R=""
+fi
```
Comment (eddelbuettel) [inline tools/cmake_call.sh:23]: A form I have seen more often is to set two vars directly such as, say, `CMAKE_ADD_AR` and have that be `CMAKE_ADD_AR="-D CMAKE_AR=$(AR)"` or else `CMAKE_ADD_AR=""` and include those below.  A little easier to read six months from now when we will have forgotten the issue.

**Review by eddelbuettel — APPROVED** [review]
Thanks for adding it.  "It works", and it would be nice if you could polish as suggested.

**Comment by tuxmaster5000** [comment]
Now the test for version 4.0.0 is used.

File: src/scripts/r_config.sh [hunk 3]
```diff
@@ -35,7 +35,7 @@ if test -z "$CXX"; then
 fi
 
 # Only R > 4 will have the variables
-${R_HOME}/bin${R_ARCH_BIN}/Rscript --vanilla -e "getRversion() > 4" | grep TRUE > /dev/null
+${R_HOME}/bin${R_ARCH_BIN}/Rscript --vanilla -e "getRversion() > '4.0.0'" | grep TRUE > /dev/null
```
Comment (eddelbuettel) [inline src/scripts/r_config.sh:38]: Decades of shell experience makes me fear I _really_ prefer out single and inner double here.  Can you revert, please?

File: tools/cmake_call.sh [hunk 4]
```diff
@@ -15,7 +15,7 @@ dot() { file=$1; shift; . "$file"; }
 dot ./scripts/r_config.sh ""
 
 # Only R > 4 will have the variables
-$RSCRIPT_BIN --vanilla -e "getRversion() > 4" | grep TRUE > /dev/null
+$RSCRIPT_BIN --vanilla -e "getRversion() > '4.0.0'" | grep TRUE > /dev/null
```
Comment (eddelbuettel) [inline tools/cmake_call.sh:18]: As above.

**Review by eddelbuettel — COMMENTED** [review]
See comments.

**Comment by tuxmaster5000** [comment]
What do mean? The code do it's job without open any security holes.

**Comment by eddelbuettel** [comment]
Our repo, our code requirements.  Please consider making the requested changes to get your PR merged.

**Comment by astamm** [comment]
What Dirk tries to convey here is that the owners of any code most of the time follow conventions when writing their code and it is good practice to follow the same rules when contributing to this code. Some go even so far as to write a code of conduct that any contributor should follow in order to contribute. We would be very grateful if you could implement Dirk's suggestions so that we can merge your PR.

**Comment by astamm** [comment]
We thank you very much for your contribution @tuxmaster5000. Can you please make the final modifications so that I can merge your PR?

**Comment by astamm** [comment]
@tuxmaster5000 can you please modify your code as per @eddelbuettel requests please ? It's been a while now. I will close this PR tomorrow in alternative and implement a fix myself. But that would be very time inefficient for everybody. Thanks.

**Pull request closed without merging**